src/components/InterpreterControl.tsx

```
@@ -0,0 +1,307 @@
+    const [isUserViewCollapsed, setIsUserViewCollapsed] =
+      useState(defaultCollapsed);
+    const [error, setError] = useState<string | null>(null);
+    const [isInitialized, setIsInitialized] = useState(false);
+
+    // Get the Formulize context to know when the instance is ready
+    // This ensures computationStore is populated before we initialize the interpreter
+    let formulizeInstance: unknown = null;
+    let formulizeIsLoading = true; // Default to loading until we know otherwise
+    let hasFormulizeContext = false;
+    try {
+      const context = useFormulize();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)